sympy/core/power.py

```diff
@@ -1142,6 +1143,10 @@ def matches(self, expr, repl_dict={}, old=False):
                 return sb.matches(b**(e/se), repl_dict)
             return sb.matches(expr**(1/se), repl_dict)
 
+        m = expand_power_base(self,force=True)
```

Comment:
The force=True is required because this isn't true in general (consider `sqrt(-1)*sqrt(-1)` vs. `sqrt(-1*-1)`). I'm worried this would result in wrong answers, if someone assumes that the match is mathematically equivalent to the pattern, but it isn't.